src/app/containers/MostRead/Canonical/index.jsx

```diff
@@ -56,19 +60,27 @@ const CanonicalMostRead = ({
 
   useEffect(() => {
     if (!items) {
-      const handleResponse = async response => {
+      const handleResponse = url => async response => {
         if (!response.ok) {
-          throw Error(response.statusText);
+          throw Error(
+            `Unexpected response (HTTP status code ${response.status}) when requesting ${url}`,
+          );
         }
         const mostReadData = await response.json();
         setItems(processMostRead({ data: mostReadData, numberOfItems }));
       };
 
       const fetchMostReadData = pathname =>
         fetch(pathname, { mode: 'no-cors' })
-          .then(handleResponse)
-          .catch(e => logger.error(`HTTP Error: "${e}"`));
+          .then(handleResponse(pathname))
+          .catch(error => {
+            logger.error(MOST_READ_FETCH_ERROR, {
+              url: pathname,
+              error,
+            });
+          });
 
+      logger.info(MOST_READ_REQUEST_RECEIVED, { url: endpoint });
```

Comment:
I think we might have to move this within the `handleResponse` function, probably after the `setItems` because currently the logger info will still be called if there is an error in the fetchMostReadData

Reply:
my bad, I thought the purpose of this log was to see when data is received. however, it is for when the fetch call is being made.